### PR TITLE
Docs for all Rust services

### PR DESCRIPTION
**Pull request by legendofa**
### DoD 
- [x] Requirements of the issue: (docs) are met
- [x] Code has been reviewed

### CODE REVIEW

File: docs/docs/dev-manuals/services/wishlist.mdx [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import CustomImage from "@site/src/components/CustomImage";
+
+# Wishlist Service
+
+The wishlist service provides the bounded context `Wishlist`.
+It is responsible for managing wishlists of users.
+
+## Domain Model
+
+// TODO: Niklas
```
Comment (delvh) [inline docs/docs/dev-manuals/services/wishlist.mdx]: What exactly is @nk-coding supposed to do?

File: docs/docs/dev-manuals/services/wishlist.mdx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  - `product_variant_connection.rs` GraphQL connection of product variants
+  - `query.rs` GraphQL queries and associated functions
+  - `user.rs` User with additional GraphQL field(s) provided by this service
+  - `wishlist_connection.rs` GraphQL connection of wishlist
+  - `wishlist.rs` Wishlist datatype
+</div>
+
+## Defined Events
+
+This section lists events that are defined by the address service and can be used by other services.
```
Comment (delvh) [inline docs/docs/dev-manuals/services/wishlist.mdx]:
```suggestion
This section lists events that are defined by the wishlist service and can be used by other services.
```

File: docs/docs/dev-manuals/services/wishlist.mdx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+n/a
+
+### Subscribed Defined Events
+
+n/a
+
+## Imported Events
+
+This section lists events defined by other services that are used by the address service.
```
Comment (delvh) [inline docs/docs/dev-manuals/services/wishlist.mdx]:
```suggestion
This section lists events defined by other services that are used by the wishlist service.
```

File: docs/docs/dev-manuals/services/wishlist.mdx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+### Published Imported Events
+
+n/a
+
+### Subscribed Imported Events
+
+- [user/user/created](/docs/docs/dev-manuals/services/user#userusercreated): Used to validate that wishlists belong to an existing users
```
Comment (delvh) [inline docs/docs/dev-manuals/services/wishlist.mdx]:
```suggestion
- [user/user/created](/docs/docs/dev-manuals/services/user#userusercreated): Used to validate that wishlists belong to an existing user
```

File: docs/docs/dev-manuals/services/wishlist.mdx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+- [catalog/product-variant/created](/docs/docs/dev-manuals/services/catalog#catalogproduct-variantcreated): Used to validate product variants of in wishlists
```
Comment (delvh) [inline docs/docs/dev-manuals/services/wishlist.mdx]:
```suggestion
- [catalog/product-variant/created](/docs/docs/dev-manuals/services/catalog#catalogproduct-variantcreated): Used to validate that product variants contained in wishlists exist
```

File: docs/docs/dev-manuals/services/wishlist.mdx (hunk 1 above)
Comment (nk-coding) [reply docs/docs/dev-manuals/services/wishlist.mdx]: it means that I agreed a while ago on drawing the hylimo diagrams for the other services if necessary and I did so with a7cfffc929ee8986ec156d9d3ce7e002ec132e21

**Comment by legendofa** [comment]
I could not start the development server, as i am getting:

```
[WARNING] No docs found in "graphql/address": can't auto-generate a sidebar.
[WARNING] No docs found in "graphql/catalog": can't auto-generate a sidebar.
[WARNING] No docs found in "graphql/discount": can't auto-generate a sidebar.
[WARNING] No docs found in "graphql/inventory": can't auto-generate a sidebar.
[WARNING] No docs found in "graphql/notification": can't auto-generate a sidebar.
[WARNING] No docs found in "graphql/return": can't auto-generate a sidebar.
[WARNING] No docs found in "graphql/review": can't auto-generate a sidebar.
[WARNING] No docs found in "graphql/shipment": can't auto-generate a sidebar.
[WARNING] No docs found in "graphql/shoppingcart": can't auto-generate a sidebar.
[WARNING] No docs found in "graphql/tax": can't auto-generate a sidebar.
[WARNING] No docs found in "graphql/user": can't auto-generate a sidebar.
[WARNING] No docs found in "graphql/wishlist": can't auto-generate a sidebar.
[WARNING] No docs found in "graphql/schema": can't auto-generate a sidebar.
[ERROR] Loading of version failed for version current
```
I think this is because of the auto-generated GraphQL docs

File: docs/docs/dev-manuals/services/wishlist.mdx [hunk 6]
```diff
@@ -33,25 +31,19 @@ The service provides a GraphQL API, which is documented in the [GraphQL API docu
...
-  - `wishlist_connection.rs` GraphQL connection of wishlist
-  - `wishlist.rs` Wishlist datatype
+ - `/src`: Source code of the service - `authentication.rs` Header authentication logic of service -
+ `base_connection.rs` Generic GraphQL connection - `foreign_types.rs` Datatypes which are not part of the services
+ bounded context - `http_event_service.rs` Handling of incoming events - `main.rs` Service execution and schema
+ generation - `mutation_input_structs.rs` GraphQL input datatypes for mutations - `mutation.rs` GraphQL mutations and
+ associated functions - `order_datatypes.rs` GraphQL datatypes for connection element ordering -
+ `product_variant_connection.rs` GraphQL connection of product variants - `query.rs` GraphQL queries and associated
+ functions - `user.rs` User with additional GraphQL field(s) provided by this service - `wishlist_connection.rs` GraphQL
+ connection of wishlist - `wishlist.rs` Wishlist datatype
```
Comment (delvh) [inline docs/docs/dev-manuals/services/wishlist.mdx]: The formatting seems to have gone haywire, please restore the old value

File: docs/docs/dev-manuals/services/invoice.mdx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+- **Framework**: [axum](https://docs.rs/axum/latest/axum/)
+- **GraphQL Library**: [async_graphql](https://docs.rs/async-graphql/latest/async_graphql/)
+- **Database**: MongoDB
+
+## Repository Structure
+
+The repository is structured as follows:
+
+<div className="repository-structure">
+ - `/src`: Source code of the service - `authentication.rs` Header authentication logic of service -
```
Comment (delvh) [inline docs/docs/dev-manuals/services/invoice.mdx]: The problematic structure is present in all new files apparently.
Please convert it to (nested) lists again.

**Review by delvh — CHANGES_REQUESTED** [review]
Additionally, @nk-coding needs to draw a few more diagrams

**Comment by nk-coding** [comment]
> I could not start the development server, as i am getting:
> 
> ```
> [WARNING] No docs found in "graphql/address": can't auto-generate a sidebar.
> [WARNING] No docs found in "graphql/catalog": can't auto-generate a sidebar.
> [WARNING] No docs found in "graphql/discount": can't auto-generate a sidebar.
> [WARNING] No docs found in "graphql/inventory": can't auto-generate a sidebar.
> [WARNING] No docs found in "graphql/notification": can't auto-generate a sidebar.
> [WARNING] No docs found in "graphql/return": can't auto-generate a sidebar.
> [WARNING] No docs found in "graphql/review": can't auto-generate a sidebar.
> [WARNING] No docs found in "graphql/shipment": can't auto-generate a sidebar.
> [WARNING] No docs found in "graphql/shoppingcart": can't auto-generate a sidebar.
> [WARNING] No docs found in "graphql/tax": can't auto-generate a sidebar.
> [WARNING] No docs found in "graphql/user": can't auto-generate a sidebar.
> [WARNING] No docs found in "graphql/wishlist": can't auto-generate a sidebar.
> [WARNING] No docs found in "graphql/schema": can't auto-generate a sidebar.
> [ERROR] Loading of version failed for version current
> ```
> 
> I think this is because of the auto-generated GraphQL docs

you can fix this by running `npm run update-graphql-docs`

**Comment by nk-coding** [comment]
> Additionally, @nk-coding needs to draw a few more diagrams

done
@legendofa  please review
note that I already included some of the stuff we discussed on Discord

**Comment by legendofa** [comment]
> > I could not start the development server, as i am getting:
> > ```
> > [WARNING] No docs found in "graphql/address": can't auto-generate a sidebar.
> > [WARNING] No docs found in "graphql/catalog": can't auto-generate a sidebar.
> > [WARNING] No docs found in "graphql/discount": can't auto-generate a sidebar.
> > [WARNING] No docs found in "graphql/inventory": can't auto-generate a sidebar.
> > [WARNING] No docs found in "graphql/notification": can't auto-generate a sidebar.
> > [WARNING] No docs found in "graphql/return": can't auto-generate a sidebar.
> > [WARNING] No docs found in "graphql/review": can't auto-generate a sidebar.
> > [WARNING] No docs found in "graphql/shipment": can't auto-generate a sidebar.
> > [WARNING] No docs found in "graphql/shoppingcart": can't auto-generate a sidebar.
> > [WARNING] No docs found in "graphql/tax": can't auto-generate a sidebar.
> > [WARNING] No docs found in "graphql/user": can't auto-generate a sidebar.
> > [WARNING] No docs found in "graphql/wishlist": can't auto-generate a sidebar.
> > [WARNING] No docs found in "graphql/schema": can't auto-generate a sidebar.
> > [ERROR] Loading of version failed for version current
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I think this is because of the auto-generated GraphQL docs
> 
> you can fix this by running `npm run update-graphql-docs`

I am getting the following error when running the graphql docs update, also tried it with Node v21. If it is just on my system then we can ignore it, just needed it to check how this PR looks like when running:

```

> docs@0.0.0 update-graphql-docs
> node scripts/updateGraphQLDocs.mjs

Error: 
      Unable to find any GraphQL type definitions for the following pointers:
        
          - schemas/address.graphql
          
    at prepareResult (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-tools/load/cjs/load-typedefs.js:82:15)
    at loadTypedefs (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-tools/load/cjs/load-typedefs.js:40:20)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async loadSchema (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-tools/load/cjs/schema.js:15:21)
    at async loadSchema (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-markdown/graphql/dist/loader.js:60:20)
    at async generateDocFromSchema (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-markdown/core/dist/generator.js:20:20)
    at async Command.<anonymous> (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-markdown/docusaurus/dist/index.js:57:17)
node:internal/errors:865
  const err = new Error(message);
              ^

Error: Command failed: npx docusaurus graphql-to-doc:address -f
Error: 
      Unable to find any GraphQL type definitions for the following pointers:
        
          - schemas/address.graphql
          
    at prepareResult (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-tools/load/cjs/load-typedefs.js:82:15)
    at loadTypedefs (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-tools/load/cjs/load-typedefs.js:40:20)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async loadSchema (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-tools/load/cjs/schema.js:15:21)
    at async loadSchema (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-markdown/graphql/dist/loader.js:60:20)
    at async generateDocFromSchema (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-markdown/core/dist/generator.js:20:20)
    at async Command.<anonymous> (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-markdown/docusaurus/dist/index.js:57:17)

    at checkExecSyncError (node:child_process:890:11)
    at execSync (node:child_process:962:15)
    at file:///home/julian/Documents/Git/misarch-docs/scripts/updateGraphQLDocs.mjs:7:20
    at ModuleJob.run (node:internal/modules/esm/module_job:195:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:336:24)
    at async loadESM (node:internal/process/esm_loader:34:7)
    at async handleMainPromise (node:internal/modules/run_main:106:12) {
  status: 1,
  signal: null,
  output: [
    null,
    '\n\n[INFO] Docusaurus version: 3.1.1\nNode version: v18.19.1\n',
    'Error: \n' +
      '      Unable to find any GraphQL type definitions for the following pointers:\n' +
      '        \n' +
      '          - schemas/address.graphql\n' +
      '          \n' +
      '    at prepareResult (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-tools/load/cjs/load-typedefs.js:82:15)\n' +
      '    at loadTypedefs (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-tools/load/cjs/load-typedefs.js:40:20)\n' +
      '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n' +
      '    at async loadSchema (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-tools/load/cjs/schema.js:15:21)\n' +
      '    at async loadSchema (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-markdown/graphql/dist/loader.js:60:20)\n' +
      '    at async generateDocFromSchema (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-markdown/core/dist/generator.js:20:20)\n' +
      '    at async Command.<anonymous> (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-markdown/docusaurus/dist/index.js:57:17)\n'
  ],
  pid: 72348,
  stdout: '\n\n[INFO] Docusaurus version: 3.1.1\nNode version: v18.19.1\n',
  stderr: 'Error: \n' +
    '      Unable to find any GraphQL type definitions for the following pointers:\n' +
    '        \n' +
    '          - schemas/address.graphql\n' +
    '          \n' +
    '    at prepareResult (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-tools/load/cjs/load-typedefs.js:82:15)\n' +
    '    at loadTypedefs (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-tools/load/cjs/load-typedefs.js:40:20)\n' +
    '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n' +
    '    at async loadSchema (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-tools/load/cjs/schema.js:15:21)\n' +
    '    at async loadSchema (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-markdown/graphql/dist/loader.js:60:20)\n' +
    '    at async generateDocFromSchema (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-markdown/core/dist/generator.js:20:20)\n' +
    '    at async Command.<anonymous> (/home/julian/Documents/Git/misarch-docs/node_modules/@graphql-markdown/docusaurus/dist/index.js:57:17)\n'
}

Node.js v18.19.1
```

File: docs/docs/dev-manuals/services/invoice.mdx (hunk 7 above)
Comment (legendofa) [reply docs/docs/dev-manuals/services/invoice.mdx]: This was caused by Prettier, if someone has a working instance feel free to format :)

**Review by legendofa — COMMENTED** [review]
Not sure about the invoice service yet: https://github.com/MiSArch/invoice/pull/1, but otherwise looks good

**Comment by legendofa** [comment]
> Not sure about the invoice service yet: [MiSArch/invoice#1](https://github.com/MiSArch/invoice/pull/1), but otherwise looks good

The Invoice interface is adapted to match the PR: https://github.com/MiSArch/invoice/pull/1

**Review by nk-coding — APPROVED** [review]

**Review by delvh — APPROVED** [review]
This PR was already open too long and should have been merged when the code review guide was published.
As such, I won't review it now but simply approve it

**Pull request merged**